features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularContext.cpp

```diff
@@ -192,14 +191,27 @@ bool UBLOX_AT_CellularContext::activate_profile(const char *apn,
                 if (_at.get_last_error() == NSAPI_ERROR_OK) {
                     // Activate, wait upto 30 seconds for the connection to be made
                     _at.set_at_timeout(30000);
-                    _at.cmd_start("AT+UPSDA=0,3");
+                    _at.cmd_start("AT+UPSD=" PROFILE ",6,");
                     _at.cmd_stop();
                     _at.resp_start();
                     _at.resp_stop();
                     _at.restore_at_timeout();
 
                     if (_at.get_last_error() == NSAPI_ERROR_OK) {
-                        activated = true;
+                        Timer t1;
+                        t1.start();
```

Comment:
If timer 1 will be stopped ?

Reply:
Timer 1 stop call is missing. However, the scope of t1 variable is limited to the if condition so this wont be a problem. 
I have already created a PR on ARM mbed, should I add timer stop call

Reply:
@fahim-ublox timer stop call is added after exiting while loop